From: Simon McVittie <smcv@collabora.com>
Date: Wed, 26 Sep 2018 13:45:30 +0100
Subject: Autotools: Always run glib-genmarshal, even if cross-compiling

glib-genmarshal is now written in Python rather than C, so we can
always run it directly, even if we're cross-compiling for an
architecture whose binaries we can't run. This matches the behaviour
of the Meson build system.

Bug: https://gitlab.gnome.org/GNOME/glib/issues/1546
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=908334
Forwarded: https://gitlab.gnome.org/GNOME/glib/merge_requests/362
Applied-upstream: 2.59.0, commit:129512b67c6329528447d05c94450b2f71e82cc3
Signed-off-by: Simon McVittie <smcv@collabora.com>
(cherry picked from commit c9f5eefa105b7e6f44f30ecc084d59b569c3dd2f)
---
 tests/gobject/Makefile.am | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/tests/gobject/Makefile.am b/tests/gobject/Makefile.am
index 0e60987..1bcefbe 100644
--- a/tests/gobject/Makefile.am
+++ b/tests/gobject/Makefile.am
@@ -51,24 +51,20 @@ if ENABLE_TIMELOOP
 installed_test_programs += timeloop-closure
 endif
 
-# The marshal test requires running a binary, which means we cannot
-# build it when cross-compiling
-if !CROSS_COMPILING
 glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal
 
 testmarshal.h: stamp-testmarshal.h
 	@true
 stamp-testmarshal.h: testmarshal.list $(glib_genmarshal)
-	$(AM_V_GEN) $(glib_genmarshal) --prefix=test_marshal $(srcdir)/testmarshal.list --header >> xgen-gmh \
+	$(AM_V_GEN) $(PYTHON) $(glib_genmarshal) --prefix=test_marshal $(srcdir)/testmarshal.list --header >> xgen-gmh \
 	&& (cmp -s xgen-gmh testmarshal.h 2>/dev/null || cp xgen-gmh testmarshal.h) \
 	&& rm -f xgen-gmh xgen-gmh~ \
 	&& echo timestamp > $@
 testmarshal.c: testmarshal.h testmarshal.list $(glib_genmarshal)
-	$(AM_V_GEN) (echo "#include \"testmarshal.h\""; $(glib_genmarshal) --prefix=test_marshal $(srcdir)/testmarshal.list --body) >> xgen-gmc \
+	$(AM_V_GEN) (echo "#include \"testmarshal.h\""; $(PYTHON) $(glib_genmarshal) --prefix=test_marshal $(srcdir)/testmarshal.list --body) >> xgen-gmc \
 	&& cp xgen-gmc testmarshal.c \
 	&& rm -f xgen-gmc xgen-gmc~
 
 BUILT_SOURCES += testmarshal.h testmarshal.c
 CLEANFILES += stamp-testmarshal.h testmarshal.h testmarshal.c
 EXTRA_DIST += testcommon.h testmarshal.list
-endif # !CROSS_COMPILING
\ No newline at end of file
